File: Brute-force/카펫.js
function solution(brown, red) {
    let answer = [];

    let inner_width = 0;
    let inner_height = 0;

    for (let i = 1; i < red + 1; i++) {
        if (red % i === 0) {
            inner_height = i
            inner_width = red / inner_height;

            if ((2 * inner_height + 2 * inner_width + 4) === brown) {
                answer.push(inner_width + 2);
                answer.push(inner_height + 2);
                break;
            }
        }
    }

    return answer;
}

File: using_cpp/BruteForce_BackTracking/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int red) {

    int len = brown / 2 + 2;

    int width = len - 3;
    int height = 3;

    while(width >= height){
        if(width * height == (brown + red)) break;

        width--;
        height++;
    }
    answer = {width, height};
    return answer;
}